code/14-location-figures.R

```diff
@@ -114,94 +92,27 @@ pop_agg = terra::aggregate(reclass$pop, fact = 20, fun = sum, na.rm = TRUE)
-         # par.settings = list(axis.line = list(col = 'transparent')), 
-         colorkey = list(space = "top", width = 0.5, height = 0.4,
-                         # make tick length shorter
-                         tck = 0.5,
-                         labels = list(cex = 0.4, at = cuts,
-                                       labels = cuts / 1000),
-                         # draw a box and ticks around the legend
-                         axis.line = list(col = "black")),
-         # legend necessary if we do not use main as legend title
-         # legend = list(top = list(fun = grid::textGrob("Size\n(m)", x = 1.05))),
-         # at command necessary again as we have a continous variable!!!
-         at = cuts,
-         # overlay with further spatial objects
-         sp.layout = list(
-           list("sp.polygons", as(ger, "Spatial"), col = gray(0.5),
-                first = FALSE),
-           list("sp.polygons", as(metros, "Spatial"), col = "gold",
-                lwd = 2, first = FALSE)
-           # list("sp.text", coords, txt = metro_names, cex = 0.7, font = 3,
-           #      first = FALSE)
-         ))
-
-# use shadow text
-# take care, latticeExtra::layer uses NSE, you need to use the data-argument!!!
-# See ?layer and:
-# browseURL(paste0("https://procomun.wordpress.com/2013/04/24/", 
-#                  "stamen-maps-with-spplot/))
-# browseURL(paste0("https://gist.github.com/oscarperpinan/",
-#                  "7482848#file-stamenpolywithlayerinfunction4-r"))
-# shadowtext by Barry Rowlingson
-# browseURL(paste0("http://blog.revolutionanalytics.com/2009/05/",
-#                  "make-text-stand-out-with-outlines.html"))
-# needs some adjustment for lattice, strwidth and strheight replace by
-# stringWidth and stringHeight (gridExtra), see
-# browseURL("https://stat.ethz.ch/pipermail/r-help/2004-November/061255.html")
-theta = seq(pi / 4, 2 * pi, length.out = 8)
-xy = xy.coords(coords)
-xo = 75 * convertWidth(stringWidth("A"), unitTo = "native", valueOnly = TRUE)
-yo = 75 * convertWidth(stringHeight("A"), unitTo = "native", valueOnly = TRUE)
-p_3 = p_2 + 
-  latticeExtra::layer(
-    for (i in theta) {
-      ltext(x = xy$x + cos(i) * xo, y = xy$y + sin(i) * yo, 
-            labels = metro_names, col = "white", font = 3, cex = 0.5)
-    },
-    data = list(xy = xy, metro_names = metro_names, theta = theta,
-                xo = xo, yo = yo)
-  ) +
-  latticeExtra::layer(
-    ltext(x = xy$x, y =  xy$y, labels = metro_names, col = "black", cex = 0.5, 
-          font = 3),
-    data = list(xy = xy, metro_names = metro_names)
-  )
-  
-# save the output
-ggplot2::ggsave(filename = "figures/08_metro_areas.png",
-                plot = arrangeGrob(p_3, ncol = 1),
-                width = 3, height = 4)
+metros$names = c("Hamburg", "Berlin", "Düsseldorf", "Leipzig",
+                 "Frankfurt am Main", "Nürnberg", "Stuttgart", "München")
+
+metros_points = st_centroid(metros)
+
+tm_2 = tm_shape(pop_agg/1000) +
```

Comment:
and again, the same comment as above, the **tmap** code looks much more concise and better to follow! Thanks a lot! I need to have  a look at the new figures!

Reply:
would it be possible to print the metro names in italics and with a smaller font?

Reply:
I made the font smaller and, at the same time, discovered a bug in {tmap} preventing me from printing the names in italics (https://github.com/r-tmap/tmap/issues/686).

Reply:
:+1: